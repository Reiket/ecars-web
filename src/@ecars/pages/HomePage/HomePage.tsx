import type {FC} from 'react';
import {Fragment} from 'react';
import {Intro} from '@ecars/uiKit/Intro';
import {InquiryCTA} from '@ecars/uiKit/InquiryCTA';
import {WhyChoose} from '@ecars/uiKit/WhyChoose';
import {Works} from '@ecars/uiKit/Works';
import {Send} from '@ecars/uiKit/Send';

export const HomePage: FC = () => (
  <Fragment>
    <Intro />
    <InquiryCTA />
    <WhyChoose />
    <Works />
    <Send />
  </Fragment>
);
