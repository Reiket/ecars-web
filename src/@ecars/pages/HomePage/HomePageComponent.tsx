import type {FC} from 'react';
import {Intro} from '@ecars/uiKit/Intro';
import {InquiryCTA} from '@ecars/uiKit/InquiryCTA';
import {WhyChoose} from '@ecars/uiKit/WhyChoose';
import {Works} from '@ecars/uiKit/Works';
import {Send} from '@ecars/uiKit/Send';
import {HomePage} from '@ecars/pages/HomePage/index';

export const HomePageComponent: FC = () => (
  <HomePage.Wrapper>
    <Intro />
    <InquiryCTA />
    <WhyChoose />
    <Works />
    <Send />
  </HomePage.Wrapper>
);
