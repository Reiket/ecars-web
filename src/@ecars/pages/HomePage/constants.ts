import {withBlockClass} from 'ecars-web-lib';
import {HomePageWrapper} from '@ecars/pages/HomePage/components/HomePageWrapper';

export const HomePageWrapperHOC = withBlockClass(HomePageWrapper, "home-page")