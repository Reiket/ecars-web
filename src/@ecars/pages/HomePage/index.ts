import {HomePageComponent} from '@ecars/pages/HomePage/HomePageComponent';
import {HomePageWrapperHOC} from '@ecars/pages/HomePage/constants';

export const HomePage = Object.assign(HomePageComponent, {
  Wrapper: HomePageWrapperHOC,
})