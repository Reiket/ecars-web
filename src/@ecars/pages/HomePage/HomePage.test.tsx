import {render} from '@testing-library/react';
import {HomePageComponent} from '@ecars/pages/HomePage/HomePageComponent';

describe('HomePage Component', () => {
  test('render component correctly', () => {
    const {container} = render(<HomePageComponent />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
