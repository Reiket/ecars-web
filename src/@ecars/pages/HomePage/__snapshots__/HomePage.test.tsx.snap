// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePageComponent Component > render component correctly 1`] = `
<div>
  <div
    class="intro"
  >
    <div
      class="intro-block"
    >
      <section
        class="intro-block__section section"
      >
        <div
          class="intro-block__container container container--lg"
        >
          <div
            class="intro-block__body"
          >
            <h1
              class="intro-block__title page-title"
            >
              Car import services with delivery to your doorstep.
            </h1>
            <div
              class="intro-block__block"
            >
              <button
                class="intro-block__button button button--big button--green"
                data-testid="buttonTestId"
                type="button"
              >
                Get a quote
              </button>
              <ul
                class="intro-block__social social"
              >
                <li
                  class="social__item"
                >
                  <a
                    class="intro-block__link link link--dark link--icon"
                    data-testid="routerLinkTestId"
                    href="https://wa.me/971558119024"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M260.062 32C138.605 32 40.134 129.701 40.134 250.232c0 41.23 11.532 79.79 31.559 112.687L32 480l121.764-38.682c31.508 17.285 67.745 27.146 106.298 27.146C381.535 468.464 480 370.749 480 250.232 480 129.701 381.535 32 260.062 32zm109.362 301.11c-5.174 12.827-28.574 24.533-38.899 25.072-10.314.547-10.608 7.994-66.84-16.434-56.225-24.434-90.052-83.844-92.719-87.67-2.669-3.812-21.78-31.047-20.749-58.455 1.038-27.413 16.047-40.346 21.404-45.725 5.351-5.387 11.486-6.352 15.232-6.413 4.428-.072 7.296-.132 10.573-.011 3.274.124 8.192-.685 12.45 10.639 4.256 11.323 14.443 39.153 15.746 41.989 1.302 2.839 2.108 6.126.102 9.771-2.012 3.653-3.042 5.935-5.961 9.083-2.935 3.148-6.174 7.042-8.792 9.449-2.92 2.665-5.97 5.572-2.9 11.269 3.068 5.693 13.653 24.356 29.779 39.736 20.725 19.771 38.598 26.329 44.098 29.317 5.515 3.004 8.806 2.67 12.226-.929 3.404-3.599 14.639-15.746 18.596-21.169 3.955-5.438 7.661-4.373 12.742-2.329 5.078 2.052 32.157 16.556 37.673 19.551 5.51 2.989 9.193 4.529 10.51 6.9 1.317 2.38.901 13.531-4.271 26.359z"
                      />
                    </svg>
                    +971 55 811 9024
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="intro-block__link link link--dark link--icon"
                    data-testid="routerLinkTestId"
                    href="mailto:request@example.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                      />
                    </svg>
                    request@example.com
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="benefits"
    >
      <section
        class="benefits__section section"
      >
        <div
          class="benefits__container container container--lg"
        >
          <div
            class="benefits__body"
          >
            <ul
              class="benefits__list"
            >
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/01.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Professional approach to clients
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/02.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Protect all payments
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/01.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Real reviews from clients
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </div>
  <div
    class="inquiry-cta"
  >
    <section
      class="inquiry-cta__section section"
    >
      <div
        class="inquiry-cta__container container container--sm"
      >
        <div
          class="inquiry-cta__body"
        >
          <div
            class="inquiry-cta__block"
          >
            <h2
              class="inquiry-cta__title block-title"
            >
              Send an inquiry now and get the best offers suitable for your requirements
            </h2>
            <p
              class="inquiry-cta__text"
            >
              Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.
            </p>
          </div>
          <button
            class="inquiry-cta__button button button--big button--green"
            data-testid="buttonTestId"
            type="button"
          >
            Get a quote
          </button>
        </div>
      </div>
    </section>
  </div>
  <div
    class="why-choose"
  >
    <section
      class="why-choose__section section"
    >
      <div
        class="why-choose__container container container--lg"
      >
        <div
          class="why-choose__body"
        >
          <div
            class="why-choose__image"
          >
            <img
              alt="why_choose_image"
              src="/assets/images/why/01.png"
            />
          </div>
          <div
            class="why-choose__content"
          >
            <h2
              class="why-choose__title section-title"
            >
              Why choose us
            </h2>
            <p
              class="why-choose__text"
            >
              Quis blandit turpis cursus in hac. In hendrerit gravida rutrum quisque. Pellentesque habitant morbi tristique senectus et. Eget gravida cum sociis natoque. Pharetra diam sit amet nisl suscipit adipiscing bibendum.
            </p>
            <p
              class="why-choose__text"
            >
              Porttitor massa id neque aliquam. In fermentum posuere urna nec. Rhoncus aenean vel elit scelerisque mauris pellentesque. Nullam ac tortor vitae purus faucibus ornare suspendisse sed nisi. Consequat id porta nibh venenatis cras sed.
            </p>
            <ul
              class="why-choose__why-choose-list why-choose-list"
            >
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    class="works"
  >
    <section
      class="works__section section"
    >
      <div
        class="works__container container container--lg"
      >
        <div
          class="works__body"
        >
          <h2
            class="works__title section-title"
          >
            How it works
          </h2>
          <p
            class="works__text"
          >
            Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
          </p>
          <ol
            class="works__works-list works-list"
          >
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                1
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Find a car
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                2
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Extensive inspection
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                3
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Safe Buying
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                4
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Delivery and support
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
          </ol>
        </div>
      </div>
    </section>
  </div>
  <div
    class="send"
  >
    <section
      class="send__section section"
    >
      <div
        class="send__container container container--lg"
      >
        <div
          class="send__body"
        >
          <div
            class="send__col"
          >
            <div
              class="send__icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12.1597 16C10.1243 16 8.29182 16.8687 7.01276 18.2556C8.38039 19.3474 10.114 20 12 20C13.9695 20 15.7727 19.2883 17.1666 18.1081C15.8956 16.8074 14.1219 16 12.1597 16ZM12 4C7.58172 4 4 7.58172 4 12C4 13.8106 4.6015 15.4807 5.61557 16.8214C7.25639 15.0841 9.58144 14 12.1597 14C14.6441 14 16.8933 15.0066 18.5218 16.6342C19.4526 15.3267 20 13.7273 20 12C20 7.58172 16.4183 4 12 4ZM12 5C14.2091 5 16 6.79086 16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5ZM12 7C10.8954 7 10 7.89543 10 9C10 10.1046 10.8954 11 12 11C13.1046 11 14 10.1046 14 9C14 7.89543 13.1046 7 12 7Z"
                />
              </svg>
            </div>
            <h2
              class="send__title block-title"
            >
              Send an inquiry and our managers will offer you the best deals.
            </h2>
            <p
              class="send__text"
            >
              Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
            </p>
            <button
              class="send__button button button--big button--green"
              data-testid="buttonTestId"
              type="button"
            >
              Get a quote
            </button>
          </div>
          <div
            class="send__col"
          >
            <div
              class="send__icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <h2
              class="send__title block-title"
            >
              Do you still have questions? Contact us for answers.
            </h2>
            <p
              class="send__text"
            >
              Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
            </p>
            <button
              class="send__button button button--big button--white"
              data-testid="buttonTestId"
              type="button"
            >
              Contact Us
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
